pkg/ingress/namespace/namespace.go

```diff
@@ -22,21 +22,28 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 
 	"github.com/apache/apisix-ingress-controller/pkg/log"
 	"github.com/apache/apisix-ingress-controller/pkg/types"
 )
 
+// FIXME: Controller should be the Core Part,
+// Provider should act as "EventHandler", register there functions to Controller
```

Comment:
Will it be implemented in a follow-up PR?

Reply:
Yes. The core concept here is that the providers shouldn't interact directly with the controller's field, for example add event handlers directly to the informers.